src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+   * @param string $required_by
+   *   Required date.
+   * @param string $quote_required
+   *   Is the quote required?
+   * @param string $category
+   *   The category of the translation.
+   * @param bool $review
+   *   True is Localize and review, False just Localize.
+   *
+   * @return int
+   *   Thebigword project id.
+   */
+  public function newTranslationProject($purchase_order_number, $project_reference, $required_by, $quote_required, $category, $review = TRUE) {
+    /** @var \DateTime $required_by */
+    $required_by->setTimezone(new \DateTimeZone('UTC'));
+    $datetime = $required_by->format('Y-m-d\TH:i:s');
+    $url = new Url('tmgmt_thebigword.callback');
+    $params = [
+      'PurchaseOrderNumber' => $purchase_order_number,
+      'ProjectReference' => $project_reference,
+      'RequiredByDateUtc' => $datetime,
+      'QuoteRequired' => $quote_required ? 'true' : 'false',
+      'SpecialismId' => $category,
+      'ProjectMetadata' => [
+        ['MetadataKey' => 'CMS User Name', 'MetadataValue' => \Drupal::currentUser()->getDisplayName()],
+        ['MetadataKey' => 'CMS User Email', 'MetadataValue' => \Drupal::currentUser()->getEmail()],
+        ['MetadataKey' => 'Response Service Base URL', 'MetadataValue' => $_SERVER['HTTP_HOST']],
+        ['MetadataKey' => 'Response Service Path', 'MetadataValue' => '/' . $url->getInternalPath()],
+      ],
+    ];
+    if ($review) {
+      $params['ProjectMetadata'][] = [
+        'MetadataKey' => 'Workflow Options',
+        'MetadataValue' => 'Localize and Review',
+      ];
+    }
+    else {
+      $params['ProjectMetadata'][] = [
+        'MetadataKey' => 'Workflow Options',
+        'MetadataValue' => 'Localize Only',
+      ];
+    }
+
+    return $this->request('project', 'POST', $params);
+  }
+
+  /**
+   * Send the files to Thebigword.
+   *
+   * @param \Drupal\tmgmt\JobItemInterface $job_item
+   *   The Job.
+   */
+  private function sendFiles(JobItemInterface $job_item) {
+    $cache = \Drupal::cache('data');
+    /** @var \Drupal\tmgmt_file\Format\FormatInterface $xliff_converter */
+    $xliff_converter = \Drupal::service('plugin.manager.tmgmt_file.format')->createInstance('xlf');
+    $resource_uuids = array();
+
+    $job_item_id = $job_item->id();
+    $target_language = $job_item->getJob()->getRemoteTargetLanguage();
+    $cid = "tmgmt_thebigword:resource_id:$job_item_id:$target_language";
+    if ($cached = $cache->get($cid)) {
+      $file_id = $cached->data;
+    }
+    else {
+      $conditions = array('tjiid' => array('value' => $job_item_id));
+      $xliff = $xliff_converter->export($job_item->getJob(), $conditions);
+      $name = "JobID_{$job_item->getJob()->id()}_JobItemID_{$job_item_id}_{$job_item->getJob()->getSourceLangcode()}_{$target_language}";
+
+      $file_id = $this->uploadFileResource($xliff, $job_item->getJob(), $job_item_id, $name);
+      $this->sendPreviewUrl($job_item, $file_id, FALSE);
```

Comment:
I know they use preview for both, but sendPreviewUrl($preview = FALSE) is weird. Either make two methods or remove Preview for the method name.